p2p/discovery.py

```diff
@@ -1083,11 +1083,16 @@ def __repr__(self) -> str:
 
 @to_list
 def _extract_nodes_from_payload(
-        payload: List[Tuple[str, str, str, str]]) -> Iterator[kademlia.Node]:
+        sender: kademlia.Address,
+        payload: List[Tuple[str, str, str, str]],
+        logger: TraceLogger) -> Iterator[kademlia.Node]:
```

Comment:
I am thinking I'd like to move away from this pattern of passing loggers around, in favor of just either a function specific logger or a module level logger.